bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockBookKeeperTestCase.java

```diff
@@ -118,9 +136,14 @@ public void setup() throws Exception {
         ledgerIdGenerator = mock(LedgerIdGenerator.class);
 
         bk = mock(BookKeeper.class);
+        Whitebox.setInternalState(bk, "bookieWatcher", bookieWatcher);
+        Whitebox.setInternalState(bk, "disableEnsembleChangeFeature", mock(Feature.class));
```

Comment:
the empty object can't achieve the goal. when the `LedgerHandle` call `handleBookieFailure`, the null of `disableEnsembleChangeFeature ` will cause NullPointerException.

Reply:
if you find somewhere direct references to bk.disableEnsembleChangeFeature you should replace them with BookKeeper.getDisableEnsembleChangeFeature
using direct fields references prevents using this mocking approach.
I think we should have only one way of mocking BookKeeper class, in order to prevent confusion

//cc @sijie

Reply:
@ArvinDevel 

I think what @eolivelli saying is to change direct access of `disableEnsembleChangeFeature` to `getDisableEnsembleChangeFeture`. and mocking this method using 'when(bk.getDisableEnsembleChangeFeature).thenReturn(mock(Feature.class))`.

Reply:
oh, got it. But the tricky thing is that `LedgerHandle` use bk's direct access to its field, see https://github.com/apache/bookkeeper/blob/f5b7e08d8ba35b7173c6c9fed4221d086cd38fdb/bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java#L1555. So we change the way to getting `disableEnsembleChangeFeature` in bk or using `whiteBox` hacky way? @sijie @eolivelli

Reply:
sorry for missing evolivell's suggestions, fix it soon.